flink-connectors/flink-connector-gcp-pubsub/src/main/java/org/apache/flink/streaming/connectors/gcp/pubsub/BlockingGrpcPubSubSubscriber.java

```diff
@@ -95,12 +95,24 @@ public void acknowledge(List<String> acknowledgementIds) {
 									.addAllAckIds(splittedAckIds.f0)
 									.build();
 
-			stub.withDeadlineAfter(60, SECONDS).acknowledge(acknowledgeRequest);
-
+			acknowledgeWithRetries(acknowledgeRequest, retries);
 			splittedAckIds = splitAckIds(splittedAckIds.f1);
 		}
 	}
 
+	private void acknowledgeWithRetries(AcknowledgeRequest acknowledgeRequest, int retriesRemaining) {
+		try {
+			stub.withDeadlineAfter(timeout.toMillis(), TimeUnit.MILLISECONDS).acknowledge(acknowledgeRequest);
+		} catch (StatusRuntimeException e) {
+			if (retriesRemaining > 0) {
+				acknowledgeWithRetries(acknowledgeRequest, retriesRemaining - 1);
```

Comment:
We probably don't want to have recursion in the retry logic. Can we replace it with a for loop instead?